streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -159,18 +140,18 @@ public String queryableStoreName() {
 
         builder.addGraphNode(this.streamsGraphNode, tableNode);
 
-
-        return new KTableImpl<>(
-            builder,
-            name,
-            processorSupplier,
-            this.keySerde,
-            this.valSerde,
-            sourceNodes,
-            shouldMaterialize ? materializedInternal.storeName() : this.queryableStoreName,
-            shouldMaterialize,
-            tableNode
-        );
+        // we can inherit parent key and value serde if user do not provide specific overrides, more specifically:
+        // we preserve the key following the order of 1) materialized, 2) parent
+        // we preserve the value following the order of 1) materialized, 2) parent
```

Comment:
Nit: Why this extensive comment? If you think we need this comment, we should add it to `KStreamImpl` cases, too. It's the same pattern for all operators for which user can provide Serdes.

Reply:
This was suggested by @vvcephei , I've also added the comment in AbstractStream as stated in the ticket (I do not think this needs to be stated in public APIs as the inheritance rule so far is still best-effort and may be evolved later).

Reply:
I agree that we don't need it in public API, however, only `KTableImpl` has extensive comments -- `KStreamImpl` has very limited comment -- thus, we should add those to `KStreamImpl`, too (for consistency). For example here: https://github.com/apache/kafka/pull/5521/files/e9eecd3b54def3df6b687acd93bd0957fc2687a0#diff-5142e1d4a6410459d6bf6df98828e5afR401 (there are others, too---basically all operators that allow to specify Serdes: through, join, etc)